File: Bit/FactorialTrailingZeroes/FactorialTrailingZeros.java
class Soluton {
	public int factorialTrailingZeros(int n) {
		if(n < 0)
			return -1;
		
		int count = 0;
		for(; n >0; n /= 5)
			count += (n / 5);
		
		return count;
	}
}

File: Bit/FactorialTrailingZeroes/recursive/FactorialTrailingZeros.cc
class Soluton {
	public:
		int factorialTrailingZeros(int n) {
			if(n == 0)
				return 0;
			else if(n < 0)
				return -1;
			else 
				return n / 5 + factorialTrailingZeros(n / 5);
		}
}